daemon/Cargo.toml

```diff
@@ -34,6 +35,35 @@ bench = false
 doc = false
 required-features = ["executable-deps"]
 
+[features]
+default = ["executable-deps"]
+executable-deps = ["dep:clap", "tokio/signal", "tokio/rt-multi-thread"]
+
+[lints.rust]
+unsafe_code = "forbid"
+unused_qualifications = "warn"
+[lints.clippy]
+pedantic = { level = "warn", priority = -1 }
+nursery = { level = "warn", priority = -1 }
+allow_attributes = "warn"
+map_with_unused_argument_over_ranges = "warn"
+
+filetype_is_file = "allow" # TODO: probably relevant for this project
+needless_continue = "allow" # TODO: probably a bug that needs to be fixed
+too_long_first_doc_paragraph = "allow"
+needless_pass_by_ref_mut = "allow" # TODO: could be a bug, could be intended
+
+cast_possible_truncation = "allow"
+cast_possible_wrap = "allow"
+cast_sign_loss = "allow"
+cognitive_complexity = "allow"
+missing_const_for_fn = "allow"
```

Comment:
const functions might be neat but probably not very helpful here as they are likely still executed on runtime.